Azure MSI for cloud auto-join (#7000)
hashicorp/consul: website/source/docs/agent/cloud-auto-join.html.md

When using Virtual Machine Scale Sets the only role action needed is `Microsoft.Compute/virtualMachineScaleSets/*/read`.

> ~> **Note:** If the Consul cluster is hosted on Azure, Consul can use Managed Service Identities (MSI) to access Azure instead of an environment variable and shared client id and secret. MSI must be enabled on the VMs hosting Consul, and it is the preferred configuration since MSI prevents your Azure credentials from being stored as clear text. This feature is supported from Consul 1.7 and above.

Reviewer: ```suggestion
~> **Note:** If the Consul cluster is hosted on Azure, Consul can use Managed Service Identities (MSI) to access Azure instead of an environment variable and shared client id and secret. MSI must be enabled on the VMs hosting Consul, and it is the preferred configuration since MSI prevents your Azure credentials from being stored in Consul configuration. This feature is supported from Consul 1.7 and above.
```

Reviewer: Small suggestion here given there are likely a few different ways to configure Consul.

Author: Makes sense @pearkes thanks for the suggestion. I've committed this to the PR
